R/ys-tidy.R

```diff
@@ -265,20 +276,80 @@ ys_join <- function(left, right, ...) {
 #' 
 #' tail(ans)
 #' 
-#' @return A `yspec` object with renamed columns
+#' data <- ys_help$data()
+#' 
+#' ys_rename(data, spec, AGE:HT)
+#' 
+#' @return 
+#' A `yspec` object or `data.frame` with renamed columns, depending on what was
+#' passed as `.x`.
 #' 
 #' @seealso [ys_join()], [ys_filter()], [ys_select()]
 #' @md
 #' @export
-ys_rename <- function(.x, ...) {
-  assert_that(is_yspec(.x))
+ys_rename <- function(.x, ... ) UseMethod("ys_rename")
+
+#' @rdname ys_rename
+#' @export
+ys_rename.yspec<- function(.x, ...) {
   re <- eval_rename(expr(c(...)), as.list(.x))
   names(.x)[re] <- names(re)
   for(i in seq_along(re)) {
     .x[[re[[i]]]][["col"]] <- names(re)[[i]]
   }
   .x
 }
+#' @rdname ys_rename
+#' @export
+ys_rename.data.frame <- function(.x, .spec, ..., .title_case = TRUE, .unit = FALSE) {
+  assert_that(is_yspec(.spec))
+  re <- eval_select(expr(c(...)), .x)
+  if(isTRUE(.unit)) {
+    short <- ys_get_short_unit(.spec, title_case = .title_case)  
+  } else {
+    short <- ys_get_short(.spec, title_case = .title_case)
+  }
+  for(i in seq_along(re)) {
+    names(.x)[re[i]] <- short[names(re)[i]]
+  }
+  .x
+}
+
+#' Recode column names in a vector
+#' 
+#' Use a `yspec` object to convert a vector of column names to the short
+#' name with the unit optionally appended. 
+#' 
+#' @param col A character vector of names in `spec`. 
+#' @param spec A `yspec` object.
+#' @param ... Passed to [ys_get_short()] (when `unit` is `FALSE`) or 
+#' [ys_get_short_unit()] (when `unit` is `TRUE`).
+#' @param unit Logical indicating if the unit should be appended to the short 
+#' name.
+#' 
+#' @examples
+#' spec <- ys_help$spec()
+#' 
+#' x <- c("WT", "BAR", "SCR", "TIME", "FOO")
+#' 
+#' ys_recode(x, spec)
```

Comment:
@KatherineKayMRG `ys_recode()` renames items in a vector or column; is this what you are wanting to do?

Reply:
So it would look like:

```r
data <- mutate(data, name = ys_recode(name, spec))
```

Reply:
Yes, that's exactly what I want to do. Sorry, I was reading the yspec vignette you shared in slack while looking through this and somehow confused ys_rename and ys_recode examples. I thought this could only rename columns (as you show in section 5.2 with ys_rename and I wanted to do what you have in the last Section 4 ys_recode example.